pkg/apis/defaults/defaults.go

```diff
@@ -55,3 +57,15 @@ func setDefaultIfEmpty(attr map[string]string, k, v string) {
 		attr[string(k)] = v
 	}
 }
+
+// setDefaultKeystorePKCS12 sets the default values for the PKCS12 relevant
+// attributes. If the csiapi.KeyStorePKCS12EnableKey key is not defined, omit
+// setting defaults on the other PKCS12 keys, since they should not be present
+// in the attributes at all. If the other attributes are present, then a
+// validation error will be picked up by validation.
+func setDefaultKeyStorePKCS12(attr map[string]string) {
+	if _, ok := attr[csiapi.KeyStorePKCS12EnableKey]; ok {
+		setDefaultIfEmpty(attr, csiapi.KeyStorePKCS12FileKey, "keystore.p12")
+		setDefaultIfEmpty(attr, csiapi.KeyStorePKCS12PasswordKey, "")
```

Comment:
The default that we're setting here is empty right?

1) why bother setting an unset key to the empty value internally?
2) what's the behaviour if no keystore password is provided? How does an end-user find out what password has been used if they didn't provide one? I expected us to make the password a required field/not perform defaulting on it.

Reply:
See below